__tests__/Base.spec.ts

```diff
@@ -79,6 +79,19 @@ describe(Base, () => {
           expect(requestError.error).toBe('Not found')
         }
       })
+
+      it('throws an ErrorObject', async () => {
+        const promise = model.withRequest('fetching', Promise.reject({ foo: 'bar' }))
+
+        try {
+          await promise
+        } catch (requestError) {
+          expect(requestError).toBeInstanceOf(ErrorObject)
+          expect(requestError.payload).toEqual({ foo: 'bar' })
+          expect(requestError.error).toEqual(null)
+          expect(requestError.requestResponse).toEqual(null)
```

Comment:
what would happen if the exception isn't raised?

Reply:
I don't understand. We are explicitly raising that exception, it can't not* be raised. It would be raised with empty values in the worse case scenario.

Reply:
👌 I'm just worried that in case something made that code correct, the test would just pass.